src/main/scala/org/broadinstitute/dsde/workbench/avram/api/AvramRoutes.scala

```diff
@@ -43,6 +61,12 @@ class AvramRoutes {
     Await.result(result, Duration.Inf)
   }
 
+  private def getToken(req: HttpServletRequest) = {
+    Option(req.getHeader("Authorization")) map {
+      _.stripPrefix("Bearer ")
```

Comment:
This could do a little more validation that it is indeed a bearer token. E.g. if we get garbage authorization headers it would be good to catch that before calling Sam.
```
private def getToken(req: HttpServletRequest): Option[String] = {
  Option(req.getHeader("Authorization")) flatMap { header =>
    val split = header.split(" ")
    if (split.length == 2 && split(0) == "Bearer") Option(split(1))
    else None
  }
}
```

Reply:
It also makes sure it's a `Bearer` token and not some other Auth type like `Basic`, etc.

Main point is to do some sanity checking of the Auth header before sending the token to Sam. Does `sttp` have any utilities for parsing common headers like `Authorization`?

Reply:
https://sttp.readthedocs.io/en/latest/requests/authentication.html

Reply:
also the check on `Bearer` should be case insensitive

Reply:
@rtitle That API is for using sttp as a client. This code is getting the token from the `HttpRequestServlet` that cloud endpoints framework is providing to our endpoint function. The real question is does cloud endpoints framework support parsing the `Authorization` header. The answer is complicated and I'd have to re-read some docs to give you a full answer, but the short version is that cloud endpoints wants to do all the authn and give our function a `User` object or something like that. We want to instead implement authn as a call to sam and I don't think there's a way for us to insert that into cloud endpoint's authn flow.